sp/src/game/server/episodic/grenade_hopwire.h

```diff
@@ -152,6 +152,50 @@ class CGravityVortexController : public CBaseEntity, public CDisplacerSink
 #endif
 };
 
+#ifdef EZ2
+class CStasisVortexController : public CGravityVortexController
+{
+	DECLARE_CLASS( CStasisVortexController, CGravityVortexController );
+	DECLARE_DATADESC();
+
+public:
+	CStasisVortexController( void ) : m_flEndTime( 0.0f ), m_flRadius( 256 ), m_flStrength( 256 ) {}
+
+	static CStasisVortexController *Create( const Vector &origin, float radius, float strength, float duration, CBaseEntity *pGrenade = NULL );
+
+	void	SetThrower( CBaseCombatCharacter *pBCC ) { m_hThrower.Set( pBCC ); }
+	CBaseCombatCharacter *GetThrower() { return m_hThrower.Get(); }
+
+	//bool	CanConsumeEntity( CBaseEntity *pEnt );
+
+	void	InputDetonate( inputdata_t &inputdata ) { StartPull( GetAbsOrigin(), m_flRadius, m_flStrength, m_flEndTime ); }
+
+	void	PullThink( void );
+	void	StartPull( const Vector &origin, float radius, float strength, float duration );
+
+private:
+	void	FreezePlayersInRange( void );
+	//bool	KillNPCInRange( CBaseEntity *pVictim, IPhysicsObject **pPhysObj );
+	void    UnfreezeNPCThink( void );
+	void	UnfreezePhysicsObjectThink( void );
+
+	float	m_flEndTime;	// Time when the vortex will stop functioning
+	float	m_flRadius;		// Area of effect for the vortex
+	float	m_flStrength;	// Pulling strength of the vortex
+
+	float	m_flStartTime;		// When the vortex opened
+	float	m_flPullFadeTime;	// How long the pull fade should last
+
+	CHandle<CBaseCombatCharacter>	m_hThrower;
+
+	// PVS for PullThink()
+	byte		m_PVS[MAX_MAP_CLUSTERS/8];
+	bool		m_bPVSCreated;
+
+	COutputEvent		m_OnPullFinished;
+};
```

Comment:
I've noticed there's a lot of duplicates from `CGravityVortexController` in here. They should be removed and the relevant members of `CGravityVortexController` should be marked as public/protected